javaparser-core/src/main/java/com/github/javaparser/ast/nodeTypes/NodeWithVariables.java

```diff
@@ -109,28 +110,28 @@ default Type getElementType() {
      * <br/>For <code>int[] a[][],b[],c[][];</code> this is int[][].
      */
     @DerivedProperty
-    default Type getMaximumCommonType() {
-        return calculateMaximumCommonType(this.getVariables().stream().map(v -> v.getType()).collect(Collectors.toList()));
+    default Optional<Type> getMaximumCommonType() {
+        return calculateMaximumCommonType(getVariables().stream().map(v -> v.getType()).collect(Collectors.toList()));
     }
 
-    static Type calculateMaximumCommonType(List<Type> types) {
+    static Optional<Type> calculateMaximumCommonType(List<Type> types) {
         // we use a local class because we cannot use an helper static method in an interface
         class Helper {
             // Conceptually: given a type we start from the Element Type and get as many array levels as indicated
             // From the implementation point of view we start from the actual type and we remove how many array
             // levels as needed to get the target level of arrays
             // It returns null if the type has less array levels then the desired target
-            private Type toArrayLevel(Type type, int level) {
+            private Optional<Type> toArrayLevel(Type type, int level) {
                 if (level > type.getArrayLevel()) {
-                    return null;
+                    return Optional.empty();
```

Comment:
Do we know of cases in which it hapens? Can it happen only if something is wrong? Should we throw an exception in that case?